concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -802,56 +818,49 @@ exportBlocksToChunk ::
-    ebtc height count lastFinRecIdx = case getBlockAt of
-        GetBlockAtV0 f ->
-            f height >>= \case
-                Nothing -> return (count, lastFinRecIdx)
-                Just (serializedBlock, finalizationRecordIndexM) -> do
-                    void $ writeBlockOut serializedBlock
-                    -- In case there wasn't a @FinalizationRecord@ present in the block
-                    -- we carry over the latest seen @FinalizationIndex@.
-                    continue count height $ fromMaybe lastFinRecIdx finalizationRecordIndexM
-        GetBlockAtV1 f ->
-            f height >>= \case
-                Nothing -> return (count, 0)
-                Just serializedBlock -> do
-                    void $ writeBlockOut serializedBlock
-                    -- We simply set the last @FinalizationRecord@ index to 0,
-                    -- as they are not a concept for 'ConsensusV1'.
-                    continue count height 0
+    ebtc :: BlockHeight -> Word64 -> Maybe (FinalizationIdentifier cpv) -> m (Word64, Maybe (FinalizationIdentifier cpv))
+    ebtc height count mLastFinIdentifier =
+        getBlockAt height >>= \case
+            Nothing -> return (count, mLastFinIdentifier)
+            Just (serializedBlock, finalizationIdentifier) -> do
+                void $ writeBlockOut serializedBlock
+                continue count height (Just finalizationIdentifier)
     writeBlockOut :: BS.ByteString -> m ()
     writeBlockOut serializedBlock = do
         let len = fromIntegral $ BS.length serializedBlock
         liftIO $ do
             BS.hPut hdl $ runPut $ putWord64be len
             BS.hPut hdl serializedBlock
-    continue count height finalizationRecordIndex =
+    continue :: Word64 -> BlockHeight -> Maybe (FinalizationIdentifier cpv) -> m (Word64, Maybe (FinalizationIdentifier cpv))
+    continue count height mFinalizationIdentifier =
         if count < chunkSize
-            then ebtc (height + 1) (count + 1) finalizationRecordIndex
-            else return (count, finalizationRecordIndex)
+            then ebtc (height + 1) (count + 1) mFinalizationIdentifier
+            else return (count, mFinalizationIdentifier)
 
 -- | Export all finalization records with indices above `dbsLastFinIndex` to a chunk
 --  Note. For 'ConsensusV1' this function will not write anything to the file.
 exportFinRecsToChunk ::
-    (MonadIO m) =>
+    (MonadIO m, cpv ~ 'ConsensusParametersVersion0) =>
     -- | Handle to export to
     Handle ->
     -- | Last finalization record index
-    FinalizationIndex ->
+    Maybe (FinalizationIdentifier cpv) ->
     -- | Action for getting the finalization record at a given index.
-    GetFinalizationRecordAt cpv m ->
+    GetFinalizationAt cpv m ->
     -- | Number of exported finalization records
     m Word64
-exportFinRecsToChunk hdl finRecIdx (GetFinalizationRecordAtV0 f) = exportFinRecsFrom (0 :: Word64) (1 + finRecIdx)
+exportFinRecsToChunk hdl mFinRecIdx f =
```

Comment:
In the documentation for `sectionFinalizationCount` you say that there can be either 0 or 1 entries, but that's not true for V0 is it?